pkg/master/client_util.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	coreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+)
+
+func createNamespaceIfNeeded(c coreclient.NamespacesGetter, ns string) error {
```

Comment:
Any way to combine with the version in controller.go?

Reply:
Both are calling this private function now.

Reply:
Does this really need its own file? This is so trivial. More boilerplate than code. And being in utils.go does not make it clearer. Calling private funcs in a package is fine.

Reply:
It doesn't absolutely need its own file but it keeps clutter out of other files.

Reply:
a matter of taste. Am fine with this file, I just wouldn't have created it ;-)

Reply:
I still prefer this over making ca post hook call a private function defined in controller.go :)